src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -391,72 +393,149 @@ void AssertAutoIndexesCount()
-            }
+            var maxSubscriptionsPerDatabase = serverStore.LicenseManager.LicenseStatus.MaxNumberOfSubscriptionsPerDatabase;
+            if (maxSubscriptionsPerDatabase is not >= 0)
+                return false;
 
-            return count;
+            if (subscriptionsCount + subscriptionsNames.Count > maxSubscriptionsPerDatabase == false)
+                return false;
+
+            if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
+                return true;
+
+            throw new LicenseLimitException(LimitType.Subscriptions,
+                $"The maximum number of subscriptions per database cannot exceed the limit of: {maxSubscriptionsPerDatabase}");
         }
+
+        return false;
+    }
+
+    private bool AssertNumberOfSubscriptionsPerClusterLimits(
+        ServerStore serverStore,
+        Table items,
+        ClusterOperationContext context,
+        IReadOnlyDictionary<string, List<string>> subscriptionsNamesPerDatabase)
+    {
+        var maxSubscriptionsPerCluster = serverStore.LicenseManager.LicenseStatus.MaxNumberOfSubscriptionsPerCluster;
+        if (maxSubscriptionsPerCluster is not >= 0)
+            return false;
+
+        var clusterSubscriptionsCounts =
+            GetDatabaseNames(context)
+                .Sum(databaseName => GetSubscriptionsCountForDatabase(context.Allocator, items, databaseName, subscriptionsNamesPerDatabase[databaseName]));
+
+        var subscriptionCommandsCount = subscriptionsNamesPerDatabase.Sum(x => x.Value.Count);
+        if (clusterSubscriptionsCounts + subscriptionCommandsCount > maxSubscriptionsPerCluster == false)
+            return false;
+
+        if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
+            return true;
+
+        throw new LicenseLimitException(LimitType.Subscriptions,
+            $"The maximum number of subscriptions per cluster cannot exceed the limit of: {maxSubscriptionsPerCluster}");
     }
 
-    public static long GetSubscriptionsCountForDatabase(ByteStringContext allocator, Table items, string name)
+    private bool AssertSubscriptionRevisionFeatureLimits(ServerStore serverStore, bool includeRevisions, ClusterOperationContext context)
     {
-        long count = 0;
+        if (serverStore.LicenseManager.LicenseStatus.HasRevisionsInSubscriptions || includeRevisions == false)
+            return true;
 
-        var subscriptionPrefix = Client.Documents.Subscriptions.SubscriptionState.SubscriptionPrefix(name).ToLowerInvariant();
+        if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
+            return true;
 
+        throw new LicenseLimitException(LimitType.Subscriptions,
+            "Your license doesn't include the subscription revisions feature.");
+    }
+
+    public static int GetSubscriptionsCountForDatabase(ByteStringContext allocator, Table items, string databaseName, List<string> subscriptionNames = null)
```

Comment:
rename argument to: subscriptionNamesToExclude

Reply:
Done.